pycm/generate_random_data.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+"""This file contains a function to generate a random confusion matrix."""
+import numpy as np
+from enum import Enum
+
+
+class ClassDistributionScenario(Enum):
+    """
+    Enum to represent different scenarios for generating class percentages.
+
+    - UNIFORM: All classes have equal representation.
+    - MAJORITY_CLASS: Only one class has a majority representation, others are minority.
+    - MINORITY_CLASS: Only one class has a minority representation, others are majority.
+    """
+    
+    UNIFORM = "uniform"
+    MAJORITY_CLASS = "majority_class"
+    MINORITY_CLASS = "minority_class"
+
+
+def _generate_class_percentages(num_classes, scenario):
+    """
+    Generate class percentages based on the given scenario.
+
+    :params num_classes: number of classes.
+    :type num_classes: int
+    :params scenario: the scenario to generate percentages for.
+    :type scenario: scenario
+    :return: list of percentages for each class.
+    """
+    if num_classes < 2:
+        raise ValueError("Number of classes must be at least 2.")
+    if scenario == ClassDistributionScenario.UNIFORM:
+        # Equal percentage for all classes
+        return [100 / num_classes] * num_classes
+    elif scenario == ClassDistributionScenario.MAJORITY_CLASS:
+        # One class has a majority class, others share the rest equally
+        raw_ratio_list = [5] + [1] * (num_classes - 1)
+    elif scenario == ClassDistributionScenario.MINORITY_CLASS:
+        # One class has a minority percentage, others share the rest equally
+        raw_ratio_list = [0.2] + [1] * (num_classes - 1)
```

Comment:
Comments for this section are no longer needed, the code is itself well-represented.